rpc/service/src/service.rs

```diff
@@ -373,6 +373,15 @@ NOTE: This error usually indicates an RPC conversion error between the node and
         })
     }
 
+    async fn get_current_block_color_call(&self, _request: GetCurrentBlockColorRequest) -> RpcResult<GetCurrentBlockColorResponse> {
+        let session = self.consensus_manager.consensus().session().await;
```

Comment:
Since consensus will apply the prune guard internally (see other comments), you can use an `unguarded_session` here. You only need a guarded session when you need consistency *between* consensus calls, otherwise it's consensus's responsibility